src/Analysis/GWASUIApp/CaseControlGWAS.jsx

```diff
@@ -511,15 +511,28 @@ const CaseControlGWAS = (props) => {
         }
     };
 
+    // This function takes in list of dichotomous covariates (cohort-pair based) and other regular covariates (concept-based)
+    // and produces the generic "variables" list that can be used as input argument in different cohort-middleware endpoints.
+    function getVariablesList(dichotomousCovariates, otherCovariates) {
+        const variables = [];
+        for (let dichotomousCovariate of dichotomousCovariates) {
+            const variableItem = {'variable_type': 'custom_dichotomous', 'cohort_ids': [dichotomousCovariate.cohort1, dichotomousCovariate.cohort2] };
```

Comment:
Will this maintain the order as selected by the user? Will we want to also have the `provided_name` or something here?

Reply:
having a `provided name` here as part of `variableItem` makes sense. if we need to preserve order then I'd suggest using javascript `map()` and then spread operator (ex. `variables = [...cdVariables, ...otherVariables]`) to put the lists together.

Reply:
@kmhernan it is not preserving the order. I judged the order not to be important at this moment. But let me know if you think otherwise!

Reply:
I guess it depends on how it is ultimately used @pieterlukasse we do care about order I think in attrition table, more important when the attrition UI component is there. Less important for the csv that is included in the tar archive. @alilahrime it would be more complicated than that in the more complicated variable UI where we could add a continuous followed by a dichotomous followed by a continuous etc. So you would have to have a single list. I'm also thinking about that new UI when there is the list of variables you selected we probably want to care about order. Maybe that is not important at all in this version though and I'm just thinking too far ahead.

Reply:
@kmhernan thanks for the explanation. 

Here are my thoughts on this matter: as I see it now, the attrition table in the frontend will not use the CSV attrition table endpoint, for performance reasons. The way I'm thinking about the implementation right now is to have each new row call the _breakdown_ endpoint at the time it is added to the list/table in the UI. This is essentially what the CSV attrition table endpoint does when generating the final attrition table CSV in the backend... however, we don't want to generate all the rows again whenever just one extra row should be added in the frontend.

Long story short: I do not have a scenario yet in which we need to inform cohort-middleware about a specific order of the variables.